internal/client/client.go

```diff
@@ -438,7 +444,34 @@ func (c *CSAPI) LoginUser(t *testing.T, localpart, password string) (userID, acc
 	return userID, accessToken, deviceID
 }
 
-//RegisterUser will register the user with given parameters and
+// LoginUserWithDeviceID will log in to a homeserver on an existing device
+func (c *CSAPI) LoginUserWithDeviceID(t *testing.T, localpart, password, deviceID string) (userID, accessToken string) {
```

Comment:
We already have a login method: `LoginUser`. The reason why you don't want to use it is because you can't specify the device ID. Rather than duplicating the entire `LoginUser` function, add functional options to `LoginUser` to set the device ID. Something like:
```go
type LoginOpt func(map[string]interface{})

func WithDeviceID(deviceID string) LoginOpt {
    return func(loginBody map[string]interface{}) {
        loginBody["device_id"] = deviceID
    }
}
```

then change `LoginUser` to be:
```go
func (c *CSAPI) LoginUser(t *testing.T, localpart, password string, opts ...LoginOpt) (userID, accessToken, deviceID string) {
    // .. existing code up to creation of reqBody
    for _, opt := range opts {
        opt(reqBody)
    }
}
```

This is further described in https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis